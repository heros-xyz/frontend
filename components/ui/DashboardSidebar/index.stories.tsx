import type { Meta, StoryObj } from "@storybook/react";
import DashboardSidebar from ".";

const meta: Meta<typeof DashboardSidebar> = {
  title: "Components/Sidebar",
  component: DashboardSidebar,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
  argTypes: {
    tabValue: {
      options: ["home", "noti", "interaction", "fan", "profile"],
      control: { type: "select" },
    },
  },
};

export default meta;
type Story = StoryObj<typeof DashboardSidebar>;

export const BottomBarStory: Story = {
  args: {
    tabValue: "home",
  },
  render: (args) => {
    console.log(args);

    return <DashboardSidebar {...args} />;
  },
};
