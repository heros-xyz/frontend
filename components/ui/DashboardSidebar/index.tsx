import { Box, BoxProps, useUpdateEffect } from "@chakra-ui/react";
import { ReactNode, useEffect, useState } from "react";
import { useRouter } from "next/router";
import Link from "next/link";
import {
  HomeIcon,
  HomeActive,
  NotificationIcon,
  NotificationActive,
  InteractionsIcon,
  InteractionsActive,
  MyfanIcon,
  MyfansActive,
  ProfileIcon,
  ProfileActive,
} from "@/components/svg/Navigate";
import LogoSidebar from "@/components/svg/LogoSidebar";
import { ACTIVE_PATHS } from "@/utils/constants";
import MenuItem from "../MenuItem";

interface DashboardSidebarProps extends BoxProps {
  tabValue: string;
  role: "ATHLETE" | "FAN";
}

interface INavItem {
  id: string;
  Icon: ReactNode;
  itemName: string;
  activeIcon: ReactNode;
  show: boolean;
  path: string;
}

const DashboardSidebar: React.FC<DashboardSidebarProps> = ({
  tabValue,
  role,
  ...props
}) => {
  const router = useRouter();
  const [tab, setTab] = useState<string>(router.pathname ?? "");

  const menuList: INavItem[] = [
    {
      id: "home",
      Icon: <HomeIcon maxW={5} />,
      itemName: "Home",
      activeIcon: <HomeActive maxW={5} />,
      show: true,
      path: role === "ATHLETE" ? "/athlete" : "/fan",
    },
    {
      id: "noti",
      Icon: <NotificationIcon maxW={5} />,
      itemName: "Notification",
      activeIcon: <NotificationActive maxW={5} />,
      show: true,
      path: role === "ATHLETE" ? "/athlete/notification" : "/fan/notification",
    },
    {
      id: "interaction",
      Icon: <InteractionsIcon maxW={5} />,
      itemName: "Interactions",
      activeIcon: <InteractionsActive maxW={5} />,
      show: true,
      path: role === "ATHLETE" ? "/athlete/interactions" : "/fan/interactions",
    },

    {
      id: "fan",
      Icon: <MyfanIcon maxW={5} />,
      itemName: "My Fans",
      activeIcon: <MyfansActive maxW={5} />,
      show: role === "ATHLETE",
      path: "/athlete/my-fan",
    },
    {
      id: "profile",
      Icon: <ProfileIcon maxW={5} />,
      itemName: "My Profile",
      activeIcon: <ProfileActive maxW={5} />,
      show: true,
      path: role === "ATHLETE" ? "/athlete/my-profile" : "/fan/my-profile",
    },
  ];

  const handleChangeTab = (tab: string): void => {
    setTab(tab);
  };

  useEffect(() => {
    if (ACTIVE_PATHS.includes(router.pathname)) {
      setTab(router.pathname);
    }
  }, [router.pathname]);

  useUpdateEffect(() => {
    setTab(tabValue);
  }, [tabValue]);

  return (
    <Box
      bg="secondary"
      p={{ base: 2.5, lg: 14 }}
      w={{ base: "full", lg: "320px" }}
      minH={{ base: "59px", lg: "100%" }}
      {...props}
    >
      <Link href="/">
        <LogoSidebar
          display={{ base: "none", lg: "block" }}
          maxW={28}
          maxH={10}
          onClick={() => setTab(role === "ATHLETE" ? "/athlete" : "/fan")}
        />
      </Link>

      <Box mt={6}>
        {menuList.map(
          (item) =>
            item.show && (
              <MenuItem
                key={item.id}
                {...item}
                isActive={tab === item.path}
                handleClickItem={handleChangeTab}
              />
            )
        )}
      </Box>
    </Box>
  );
};

export default DashboardSidebar;
