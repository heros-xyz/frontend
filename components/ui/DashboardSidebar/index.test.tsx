import { render } from "@testing-library/react";
import mockRouter from "next-router-mock";
import DashboardSidebar from ".";
jest.mock("next/router", () => require("next-router-mock"));

it("renders Dashboard Sidebar unchanged", () => {
  mockRouter.push("/initial-path");
  const { container } = render(<DashboardSidebar tabValue="home" role="FAN" />);
  expect(container).toMatchSnapshot();
});
