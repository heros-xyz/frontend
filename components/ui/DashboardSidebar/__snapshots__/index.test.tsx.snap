// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Dashboard Sidebar unchanged 1`] = `
<div>
  <div
    class="css-14gkuht"
  >
    <a
      href="/"
    >
      <svg
        class="chakra-icon css-pxohan"
        focusable="false"
        viewBox="0 0 115 32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 31.3833V7.10351C0 3.17488 3.12758 0 6.99768 0V10.8266C8.39271 8.86225 10.9803 7.69737 13.9054 7.69737C19.283 7.69737 22.4106 11.192 22.4106 17.1992V31.3833H15.4579V18.9122C15.4579 16.0114 13.8829 14.1385 11.4753 14.1385C8.70772 14.1385 6.95268 15.9658 6.95268 18.8209V31.3833H0Z"
          fill="#33EFEF"
        />
        <path
          d="M25.875 19.8715C25.875 12.7224 30.6226 7.72021 37.4403 7.72021C44.528 7.72021 48.9606 12.4483 48.9606 19.9172V21.6988L32.5577 21.7902C32.9627 24.828 34.6277 26.2442 37.7103 26.2442C39.0828 26.2442 40.2529 25.9701 41.1079 25.4904C43.2904 24.257 45.7655 23.6403 48.2856 23.6403H49.0506C48.2406 28.7338 43.763 32 37.5753 32C30.7576 31.9772 25.875 26.9294 25.875 19.8715ZM32.6477 17.4276H42.0529C42.0529 15.0064 40.2979 13.4304 37.4853 13.4304C34.7177 13.4533 33.1877 14.6867 32.6477 17.4276Z"
          fill="#33EFEF"
        />
        <path
          d="M68.334 15.0978H66.1289C62.1463 15.0978 59.7613 16.8794 59.7613 21.4247V24.3255C59.7613 28.2313 56.6337 31.3833 52.8086 31.3833V8.4968H59.3563L59.7163 11.9001C60.6613 9.66169 62.6638 8.13135 65.6789 8.13135C66.4889 8.13135 67.389 8.22271 68.3565 8.45112V15.0978H68.334Z"
          fill="#33EFEF"
        />
        <path
          d="M96.0091 24.1429C98.5066 24.1429 100.914 24.9195 103.074 26.1986C103.637 26.5183 104.357 26.7011 105.279 26.7011C107.439 26.7011 108.452 25.9245 108.452 24.7368C108.452 24.0515 108.092 23.3206 106.427 22.9552L102.264 22.0415C98.1016 21.1279 96.0766 19.2778 96.0766 15.2578C96.0766 10.4841 100.104 7.74316 105.527 7.74316C110.769 7.74316 114.437 10.8724 114.437 15.4862H113.717C111.242 15.4862 108.879 14.6411 106.787 13.2935C106.382 13.0194 105.842 12.8824 105.212 12.8824C103.637 12.8824 102.624 13.7046 102.624 14.938C102.624 15.9887 103.479 16.5826 105.054 16.948L109.262 17.8617C112.975 18.6839 115 20.4198 115 24.0972C115 29.008 110.882 32.0002 105.009 32.0002C99.0467 32.0002 95.2891 28.7796 95.2891 24.1429H96.0091Z"
          fill="#33EFEF"
        />
        <path
          d="M90.9707 12.5395C88.8781 9.8443 85.6155 8.1084 81.9479 8.1084C78.5279 8.1084 75.9403 8.88499 73.8477 11.3061C72.0477 13.3618 70.4727 16.7879 70.4727 19.7572C70.4727 23.0692 71.8902 26.0842 74.0727 28.2084C76.1428 30.2183 80.2154 31.4061 83.298 31.4061C87.0556 31.4061 89.5306 28.9849 91.6232 26.1527C93.0632 24.2112 93.4232 22.3839 93.4232 19.7572C93.4232 17.0392 92.5007 14.5267 90.9707 12.5395ZM85.1205 21.9271C83.2755 24.6909 80.3504 25.9699 78.6179 24.7594C76.8628 23.5488 76.9528 20.3283 78.7979 17.5645C80.6429 14.8008 83.568 13.5217 85.3005 14.7322C87.0331 15.9428 86.9656 19.1634 85.1205 21.9271Z"
          fill="#33EFEF"
        />
      </svg>
    </a>
    <div
      class="css-19ug521"
    >
      <a
        href="/fan"
      >
        <div
          class="css-1r5k4l9"
        >
          <div
            class="css-0"
          >
            <svg
              class="chakra-icon css-v4492d"
              focusable="false"
              viewBox="0 0 18 18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.5 9L3.16667 7.33333M3.16667 7.33333L9 1.5L14.8333 7.33333M3.16667 7.33333V15.6667C3.16667 16.1269 3.53976 16.5 4 16.5H6.5M14.8333 7.33333L16.5 9M14.8333 7.33333V15.6667C14.8333 16.1269 14.4602 16.5 14 16.5H11.5M6.5 16.5C6.96024 16.5 7.33333 16.1269 7.33333 15.6667V12.3333C7.33333 11.8731 7.70643 11.5 8.16667 11.5H9.83333C10.2936 11.5 10.6667 11.8731 10.6667 12.3333V15.6667C10.6667 16.1269 11.0398 16.5 11.5 16.5M6.5 16.5H11.5"
                stroke="#33EFEF"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="1.5"
              />
            </svg>
          </div>
          <div
            class="chakra-text css-1ss9ua5"
          >
            Home
          </div>
        </div>
      </a>
      <a
        href="/fan/notification"
      >
        <div
          class="css-1r5k4l9"
        >
          <div
            class="css-0"
          >
            <svg
              class="chakra-icon css-v4492d"
              focusable="false"
              viewBox="0 0 16 18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.5 13.1667H14.6666L13.4959 11.9959C13.1784 11.6784 13 11.2477 13 10.7987V8.16667C13 5.98964 11.6086 4.13757 9.66665 3.45118V3.16667C9.66665 2.24619 8.92045 1.5 7.99998 1.5C7.0795 1.5 6.33331 2.24619 6.33331 3.16667V3.45118C4.39133 4.13757 2.99998 5.98964 2.99998 8.16667V10.7987C2.99998 11.2477 2.8216 11.6784 2.50409 11.9959L1.33331 13.1667H5.49998M10.5 13.1667V14C10.5 15.3807 9.38069 16.5 7.99998 16.5C6.61927 16.5 5.49998 15.3807 5.49998 14V13.1667M10.5 13.1667H5.49998"
                stroke="#33EFEF"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="1.5"
              />
            </svg>
          </div>
          <div
            class="chakra-text css-1ss9ua5"
          >
            Notification
          </div>
        </div>
      </a>
      <a
        href="/fan/interactions"
      >
        <div
          class="css-1r5k4l9"
        >
          <div
            class="css-0"
          >
            <svg
              class="chakra-icon css-v4492d"
              focusable="false"
              viewBox="0 0 18 18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.66667 6.33325H5.675M9 6.33325H9.00833M12.3333 6.33325H12.3417M6.5 11.3333H3.16667C2.24619 11.3333 1.5 10.5871 1.5 9.66659V2.99992C1.5 2.07944 2.24619 1.33325 3.16667 1.33325H14.8333C15.7538 1.33325 16.5 2.07944 16.5 2.99992V9.66659C16.5 10.5871 15.7538 11.3333 14.8333 11.3333H10.6667L6.5 15.4999V11.3333Z"
                stroke="#33EFEF"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="1.5"
              />
            </svg>
          </div>
          <div
            class="chakra-text css-1ss9ua5"
          >
            Interactions
          </div>
        </div>
      </a>
      <a
        href="/fan/my-profile"
      >
        <div
          class="css-1r5k4l9"
        >
          <div
            class="css-0"
          >
            <svg
              class="chakra-icon css-v4492d"
              focusable="false"
              viewBox="0 0 14 18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.3333 4.83333C10.3333 6.67428 8.84094 8.16667 6.99999 8.16667C5.15904 8.16667 3.66666 6.67428 3.66666 4.83333C3.66666 2.99238 5.15904 1.5 6.99999 1.5C8.84094 1.5 10.3333 2.99238 10.3333 4.83333Z"
                stroke="#33EFEF"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="1.5"
              />
              <path
                d="M6.99999 10.6667C3.77833 10.6667 1.16666 13.2783 1.16666 16.5H12.8333C12.8333 13.2783 10.2217 10.6667 6.99999 10.6667Z"
                stroke="#33EFEF"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="1.5"
              />
            </svg>
          </div>
          <div
            class="chakra-text css-1ss9ua5"
          >
            My Profile
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
